seat.c

```diff
@@ -314,27 +366,19 @@ handle_new_keyboard(struct cg_seat *seat, struct wlr_input_device *device)
 							   XKB_KEYMAP_COMPILE_NO_FLAGS);
 	if (!keymap) {
 		wlr_log(WLR_ERROR, "Unable to configure keyboard: keymap does not exist");
-		free(keyboard);
 		xkb_context_unref(context);
 		return;
 	}
 
-	keyboard->seat = seat;
-	keyboard->device = device;
 	wlr_keyboard_set_keymap(device->keyboard, keymap);
 
 	xkb_keymap_unref(keymap);
 	xkb_context_unref(context);
 	wlr_keyboard_set_repeat_info(device->keyboard, 25, 600);
 
-	wl_list_insert(&seat->keyboards, &keyboard->link);
-	keyboard->destroy.notify = handle_keyboard_destroy;
-	wl_signal_add(&device->events.destroy, &keyboard->destroy);
-	keyboard->key.notify = handle_keyboard_key;
-	wl_signal_add(&device->keyboard->events.key, &keyboard->key);
-	keyboard->modifiers.notify = handle_keyboard_modifiers;
-	wl_signal_add(&device->keyboard->events.modifiers, &keyboard->modifiers);
+	cg_keyboard_group_add(device, seat);
 
+	wl_signal_init(&device->events.destroy);
```

Comment:
With `handle_keyboard_destroy` removed, is this still needed?

Reply:
I don't think so... I removed it and nothing seems to be broken, so it should be find ;).